src/transformers/models/decision_transformer/modeling_decision_transformer.py

```diff
@@ -0,0 +1,224 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch DecisionTransformer model."""
+
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward
+from ...modeling_outputs import DecisionTransformerOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import logging
+from ..gpt2 import GPT2Model
```

Comment:
This breaks the core philosophy of Transformers that each model file should be fully independent from the others, so we never import other components in a modeling file. Instead we copy the blocks we need and add `# Copied from` statements.

Reply:
Agree here with @sgugger I think. Sorry @edbeeching I might have been a bit misleading in my messages on Slack. 

To be more precise, I think we should never import a specfic model like `GPT2Model` into another modeling file. I think there are two possibilities here:

1. Decision Transformers only works with GPT2 (all it's checkpoints are GPT2-like checkpoints). In this case I 100% agree with @sgugger and we should just copy-paste the whole GPT2Model architecture under the name `DecisionTransformerModel` into this file and add a nice `# Copied from ...` line to ensure the code is kept up to date.
2. If we have checkpoints for GPT2, GPT-J and GPT-Neo, then I can be convinced into importing the `AutoModelForCausalLM` class here and make the model a composition model.

However given that the model is just a thin wrapper above GPT2 (e.g. essentially just uses GPT2 as the encoder), I also prefer 1. here.


Sorry, @edbeeching I might have been misleading on Slack (first time I'm doing a full review on the PR)

Reply:
No problem. Do you have an example where this has been done before? So I do not have to write/copy/adjust too much boilerplate. Should I call it DecisionTransformerGPT2Model ? I guess I need to copy all the tests as well?